src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,152 @@
+package duke.task;
+
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.TaskDoesNotExistException;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskList {
+    public List<Task> list;
+
+    public TaskList() {
+        this.list = new ArrayList<>();
+    }
+
+    public TaskList(List<Task> list) {
+        this.list = list;
+    }
+
+    public List<Task> getTaskList() {
+        return this.list;
+    }
+
+    public int getListSize() {
+        return this.list.size();
+    }
+    public void getList() {
+        int counter = 1;
+        for (Object task : this.list) {
+            System.out.println(counter + ". " + task.toString());
+            counter++;
+        }
+    }
+
+    public void markTask(String[] parts) throws TaskDoesNotExistException {
+        //converting the index from String to Int
+        int indexInt = Integer.parseInt(parts[1]) - 1;
+
+        //in case the duke.task.Task index to mark exceeds current number of Tasks or neg number
+        if (indexInt + 1 > list.size() || indexInt < 0) {
+            throw new TaskDoesNotExistException();
+        }
+
+        Task t = list.get(indexInt);
+        t.markDone();
+        System.out.println("Nice! I've marked this duke.task as done:\n" + t.toString());
+    }
+
```

Comment:
I like that you followed method naming convention and adding comment to explain the code